File: Singleton.cpp
class Solution {
public:
    /**
     * @return: The same instance of this class every time
     */
    static Solution* getInstance() {
        // write your code here
        //copy the answer.
        static Solution *instance = new Solution();
        return instance;
    }
    
    Solution(const Solution&) = delete;
    Solution& operation(const Solution&) = delete;
private:
    Solution(){};
    ~Solution(){};
};

File: Singleton.py
import threading
class Solution:
    # @return: The same instance of this class every time
    instance = None
    mutex = threading.Lock()
    @classmethod
    def getInstance(cls):
        # write your code here
        # if cls.instance is None:
        #     cls.instance = Solution()
        # return cls.instance
        
        if cls.instance is None:
            cls.mutex.acquire()
            if cls.instance is None:
                cls.instance = cls()
            cls.mutex.release()
        
        return cls.instance